docs/SQL/development/visual-editor/visual-expression-builder/visual-expression-builder-reference.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Visual Expression Builder reference
+id: visual-expression-builder-reference
+description: Visual Expression Builder reference
+sidebar_position: 3
+tags:
+  - reference
+  - development
+  - visual
+  - functions
+  - expression
+  - sql
+---
+
+This page contains a reference of the different Visual Expression Builder components, which include the expression options, operator options, and data types.
+
+## Expression options
+
+The Visual Expression Builder supports the following expression options:
+
+- **Column**: Allows you to select an input column from your source tables. You can view all of the available input columns from under the dropdown menu or under **Input** on the left-hand side of the Gem dialog.
+- **Value**: Allows you to enter any kind of value.
+  - If you enter a string value, it'll be considered as a string within quotes.
+  - If you enter a number, it'll be considered as a numerical value, but you have the option to click to **Check to read value as string**.
+  - The same applies to a boolean value. For example, if you enter `True`, then it'll be considered a boolean value unless you **Check to read value as string**.
+- **Function**: Includes a list of all of the function category groups and functions that are supported. The list displays each function description, including mandatory arguments.
+- **Conditional**: Allows you to use a conditional `WHEN` clause.
+  - Within `WHEN`, you can select your columns and operator, and within `THEN` you can add a single simple expression.
+  - You can add multiple `CASES` of the `WHEN` clause, but you can only have one `ELSE` statement. `ELSE` also uses a simple expression.
+  - You can also add `IF`, `ELSEIF`, or `FOR` conditions between each of your expressions. `FOR` conditions take a variable name and an expression value. `IF` and `ELSEIF` conditions are considered comparisons.
```

Comment:
IF, ELSEIF and FOR are a feature of `Code table` not of Visual expression builder, this is already there in prod we just added a visual counter part to it

Reply:
This is available only in `Expressions` table in Reformat, Aggregate and Join gems

Reply:
Ack. I added a line for the Reformat, Aggregate, and Join Gems.